app/src/main/java/com/infomaniak/mail/ui/bottomSheetDialogs/ScheduleSendBottomSheetDialog.kt

```diff
@@ -0,0 +1,224 @@
+import dagger.hilt.android.AndroidEntryPoint
+import java.util.Date
+import javax.inject.Inject
+
+
+@AndroidEntryPoint
+class ScheduleSendBottomSheetDialog @Inject constructor() : ActionsBottomSheetDialog() {
+
+    private val navigationArgs: ScheduleSendBottomSheetDialogArgs by navArgs()
+
+    private var binding: BottomSheetScheduleSendBinding by safeBinding()
+    override val mainViewModel: MainViewModel by activityViewModels()
+
+    private val newMessageViewModel: NewMessageViewModel by activityViewModels()
+
+    @Inject
+    lateinit var localSettings: LocalSettings
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return BottomSheetScheduleSendBinding.inflate(inflater, container, false).also { binding = it }.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?): Unit = with(binding) {
+        super.onViewCreated(view, savedInstanceState)
+
+        localSettings.lastSelectedScheduleDate?.let { lastSelectedSchedule ->
+            if (Date(lastSelectedSchedule).isAtLeastXMinutesInTheFuture(MIN_SCHEDULE_DELAY_MINUTES)) {
+                lastScheduleItem.isVisible = true
+                lastScheduleItem.setDescription(
+                    mostDetailedDate(
+                        context,
+                        date = Date(lastSelectedSchedule),
+                        format = FORMAT_DATE_DAY_MONTH,
+                    )
+                )
+            }
+        }
+
+        lastScheduleItem.setClosingOnClickListener {
+            val draftResource = navigationArgs.draftResource
+            val lastSelectedScheduleDate = localSettings.lastSelectedScheduleDate
+
+            if (navigationArgs.isAlreadyScheduled) {
+                if (draftResource != null && lastSelectedScheduleDate != null) {
+                    trackScheduleSendEvent("lastSchedule")
+                    mainViewModel.rescheduleDraft(draftResource, Date(lastSelectedScheduleDate))
+                }
+            } else {
+                lastSelectedScheduleDate?.let {
+                    trackScheduleSendEvent("lastSchedule")
+                    newMessageViewModel.setScheduleDate(Date(it))
+                    newMessageViewModel.triggerSendMessage()
+                }
+            }
+        }
+
+        if (newMessageViewModel.currentMailbox.isFree) {
+            customScheduleItem.setOnClickListener {
+                safeNavigate(
+                    resId = R.id.upgradeProductBottomSheetDialog,
+                    currentClassName = ScheduleSendBottomSheetDialog::class.java.name,
+                )
+            }
+        } else {
+            customScheduleItem.setClosingOnClickListener {
+                if (navigationArgs.isAlreadyScheduled) {
+                    mainViewModel.showSelectDateAndTimeForScheduleDialog()
+                } else {
+                    newMessageViewModel.showSelectDateAndTimeForScheduleDialog()
```

Comment:
Using the view models as side effects to trigger the date picker is part of the reasons why this class has high coupling. I think we have situations where we returned values from bottom sheet or dialogs to the caller so he decides what to do next